src/pybind/rgw_admin_user/rgw_admin_user.pyx

```diff
@@ -0,0 +1,290 @@
+"""
+Wrap librgw_admin_user -- borrows liberally from rgw.pyx
+"""
+
+from cpython cimport PyObject, ref, exc, array
+from libc.stdint cimport *
+from libcpp cimport bool
+from libc.stdlib cimport malloc, realloc, free
+from libc.string cimport const_char, const_uchar
+
+from collections import namedtuple
+import errno
+
+cdef extern from "Python.h":
+    # These are in cpython/string.pxd, but use "object" types instead of
+    # PyObject*, which invokes assumptions in cpython that we need to
+    # legitimately break to implement zero-copy string buffers in Image.read().
+    # This is valid use of the Python API and documented as a special case.
+    PyObject *PyBytes_FromStringAndSize(char *v, Py_ssize_t len) except NULL
+    char* PyBytes_AsString(PyObject *string) except NULL
+    int _PyBytes_Resize(PyObject **string, Py_ssize_t newsize) except -1
+    void PyEval_InitThreads()
+
+cdef extern from "rgw/librgw_admin_user.h" nogil:
+    ctypedef void* librgw_admin_user_t
+
+    # setup/teardown
+    int librgw_admin_user_create(librgw_admin_user_t *rgw_admin_user, int argc,
+    	char **argv)
+    void librgw_admin_user_shutdown(librgw_admin_user_t rgw_admin_user)
+
+    ctypedef const_char specialChar
+
+    cdef struct rgw_user_info:
+        const_char *uid
+        const_char *display_name
+        const_char *access_key
+        const_char *secret_key
+        const_char *email
+        const_char *caps
+        const_char *access
+        bool admin
+        bool system
+
+    # get interface version
+    void rgw_admin_user_version(int *major, int *minor, int *extra)
+
+    # create a new rgw user
+    int rgw_admin_create_user(librgw_admin_user_t rgw_admin_user,
+    	const_char *uid, const_char *display_name,  const_char *access_key,
+	const_char *secret_key, const_char *email, const_char *caps,
+	const_char *access, bool admin, bool system)
+
+    # get rgw user info
+    int rgw_admin_user_info(librgw_admin_user_t rgw_admin_user, const_char *uid,
+    rgw_user_info *user_info)
+
+    # free user_info structure
+    void rgw_admin_user_release_info(rgw_user_info* user_info)
+
+UserInfo = namedtuple('UserInfo', ["uid", "display_name", "access_key", "secret_key", "email", "caps", "access", "admin", "system"])
+    
+class Error(Exception):
```

Comment:
Just curious: why do we need to copy the exceptions all over again? As the exceptions are everywhere completely identical, would it make sense to just have them at one location?

Reply:
@sebastian-philipp i think the reason is that we still don't have a package which is shared across different python bindings. we don't have a package listed as a dependency of all python binding packages. i think the mgr modules also suffer from the same problem -- how to share/reuse the common bits.

Reply:
The mgr modules can share common code in the pybind/mgr dir, but there is indeed no real solution for code common between ceph components.

Reply:
ahh, right! i just realized that we have extracted `mgr_util.py`. IMHO, `mgr_module.py` is kinda awkward as we are dumping more and more utilities into it.

Reply:
See #28837 a solution to this:
```suggestion
from ceph import exception
```

Reply:
yeah, this should now start to work. Might need to make `python-ceph-common` available in this context, though.